splink/internals/linker_components/inference.py

```diff
@@ -734,67 +761,76 @@ def compare_two_records(
                     )
 
         sql_join_tf = _join_new_table_to_df_concat_with_tf_sql(
-            self._linker, "__splink__compare_two_records_left"
+            linker, "__splink__compare_two_records_left", df_records_left
         )
 
         pipeline.enqueue_sql(sql_join_tf, "__splink__compare_two_records_left_with_tf")
 
         sql_join_tf = _join_new_table_to_df_concat_with_tf_sql(
-            self._linker, "__splink__compare_two_records_right"
+            linker, "__splink__compare_two_records_right", df_records_right
         )
 
         pipeline.enqueue_sql(sql_join_tf, "__splink__compare_two_records_right_with_tf")
 
-        source_dataset_ic = (
-            self._linker._settings_obj.column_info_settings.source_dataset_input_column
-        )
-        uid_ic = self._linker._settings_obj.column_info_settings.unique_id_input_column
-
         pipeline = add_unique_id_and_source_dataset_cols_if_needed(
-            self._linker,
+            linker,
             df_records_left,
             pipeline,
             in_tablename="__splink__compare_two_records_left_with_tf",
             out_tablename="__splink__compare_two_records_left_with_tf_uid_fix",
             uid_str="_left",
         )
         pipeline = add_unique_id_and_source_dataset_cols_if_needed(
-            self._linker,
+            linker,
             df_records_right,
             pipeline,
             in_tablename="__splink__compare_two_records_right_with_tf",
             out_tablename="__splink__compare_two_records_right_with_tf_uid_fix",
             uid_str="_right",
         )
 
-        sqls = block_using_rules_sqls(
```

Comment:
If we're using cartesian blocking, we don't need to run any complex blocking code.

In addition, this code creates and materilises a list of pairwise Ids, which is uses for the join.  This is unnecessary in the context of a handful of records